Source/JavaScriptCore/bytecode/InlineCacheCompiler.cpp

```diff
@@ -4343,7 +4441,7 @@ AccessGenerationResult InlineCacheCompiler::regenerate(const GCSafeConcurrentJSL
     m_scratchFPR = scratchFPR.value_or(InvalidFPRReg);
     m_doesCalls = doesCalls;
     m_doesJSCalls = doesJSCalls;
-    if (needsSymbolPropertyCheck || needsStringPropertyCheck || needsInt32PropertyCheck)
+    if (needsSymbolPropertyCheck || needsStringPropertyCheck || needsInt32PropertyCheck || doesJSCalls)
```

Comment:
Nit add the comment from above: `Cases that do JS calls rely on stack-pointer adjustment, which is tied to the linked CodeBlock, which makes this code unshareable.`

Reply:
In fact, this comment was stale and no longer true :) We are storing this information in JITData and we can adjust it without using CodeBlock embedded pointer (we are already doing that for Baseline JIT IC).
The current disablement is largely because CallLinkInfo pointer is embedded into the code itself. (And AccessCase holds it).